src/firestore/indexes-api.ts

```diff
@@ -65,5 +65,5 @@ export interface Field {
  */
 export interface IndexConfig {
   ancestorField?: string;
-  indexes: Index[];
```

Comment:
I thought the API would return `indexes: []` in this case but it's really `indexes: undefined`.  Changed the type here, fixed all call sites, added a test to make sure we consider this equivalent.